politeiawww/invoices.go

```diff
@@ -927,6 +927,55 @@ func (p *politeiawww) processEditInvoice(ei cms.EditInvoice, u *user.User) (*cms
 	}, nil
 }
 
+// processGeneratePayouts looks for all approved invoices and uses the provided
+// exchange rate to generate a list of addresses and amounts for an admin to
+// process payments.
+func (p *politeiawww) processGeneratePayouts(gp cms.GeneratePayouts, u *user.User) (*cms.GeneratePayoutsReply, error) {
+	log.Tracef("processGeneratePayouts")
+
+	dbInvs, err := p.cmsDB.InvoicesByStatus(int(cms.InvoiceStatusApproved))
+	if err != nil {
+		return nil, err
+	}
+
+	// XXX ? Should we trust the dbInvs or should we take the token and get the
```

Comment:
I would just use the cmsdb response.  Is there a reason that it shouldn't be trusted?

Reply:
Maybe previously, but I believe we're nearing the point where cmsdb is fully recoverable from the raw records.  So much less of a concern now.